srtcore/api.cpp

```diff
@@ -2590,7 +2590,7 @@ void CUDTUnited::updateListenerMux(CUDTSocket* s, const CUDTSocket* ls)
    for (map<int, CMultiplexer>::iterator i = m_mMultiplexer.begin();
       i != m_mMultiplexer.end(); ++ i)
    {
-      if (i->second.m_iPort == port)
+      if (i->second.m_iPort == port && i->second.m_iID == ls->m_SocketID)
```

Comment:
```suggestion
      // In case of multiple listeners on the same UDP port it is important to also
      // find the multiplexer by its ID. This ID should correspond to the exact listener socket,
      // that accepted the connection.
      // It is done by comparing socket ID
      if (i->second.m_iPort == port && i->second.m_iID == ls->m_SocketID)
```